src/onion_messenger.rs

```diff
@@ -199,18 +199,29 @@ impl LndkOnionMessenger {
         let peers_sender = sender.clone();
         let (peers_shutdown, peers_listener) = (signals.shutdown.clone(), signals.listener.clone());
         set.spawn(async move {
-            let peer_subscription = peers_client
+            let peer_subscription = match peers_client
                 .subscribe_peer_events(tonic_lnd::lnrpc::PeerEventSubscription {})
                 .await
-                .expect("peer subscription failed")
-                .into_inner();
+            {
+                Ok(response) => {
+                    info!("Connected to peer events subscription.");
+                    response.into_inner()
+                }
+                Err(e) => {
+                    peers_shutdown.trigger();
+                    error!("Error subscribing to peer events: {e}.");
+                    return;
+                }
+            };
 
             let peer_stream = PeerStream {
                 peer_subscription,
-                client: peers_client,
+                client: peers_client.clone(),
```

Comment:
Let's be careful about using .clone(). It looks like this .clone, as well as the clones in line 222 and 260 + 261 aren't needed

Reply:
oh my bad, I left if from my previous "loop solution". it's done!

Reply:
btw, I did just amend the changes as the lines are clearly stated on your comment 🙏🏼

Reply:
should probably set up a clippy ci for some basic stuff

Reply:
Weird, yeah you'd think it would check for that